### DISCUSSION TITLE
asserting an error event in xstate5?

### DISCUSSION THREAD

**Discussion by emarksamica** [statelyai/xstate#4735] [General]
I want to define a reusable action that assigns the error response from a failed invoke to context:

```
        [Actions.ASSIGN_ERROR_TO_CONTEXT]: assign({
          APIError: ({event}) => event.error
        })
```

TypeScript balks at this though because event is typed as being of the events I've declared in setup({types: events}). Is there a means to assertEvent so TS infers the type of the event as an invocation onError? At the moment I've only found @ts-ignore as a work around.

**Comment by crishoj** [comment]
@emarksamica did you find a solution here?

**Comment by taraldr** [comment]
If I understand the issue correct, this might be a solution.

You can define a type for actor invocation errors with a generic for the error type like so:

```ts
type ActorOnErrorEvent = <
    TType extends `xstate.error.actor.${string}`,
    TErrorData,
> = {
    type: TType;
    error: TErrorData;
};
```

use it 

```ts
type Events = ActorOnErrorEvent<"xstate.error.actor.idOfActor", { foo: string; }> | //...;
```

and then assert the event like so 

```ts
onDone: [
    {
        type: Actions.ASSIGN_ERROR_TO_CONTEXT,
        params: ({ event }) => {
            assertEvent(
                event,
                "xstate.error.actor.idOfActor",
            );
            return {
                APIError: event.error, // Error is now `{ foo: string; }`
            };
        },
    },
],
```
